### PR TITLE
Migrate to Material Symbols

### LINKED ISSUE #18037: Migrate to Material "Symbols"
### Discussed in https://github.com/directus/directus/discussions/12958

<div type='discussions-op-text'>

<sup>Originally posted by **LasseRosenow** April 22, 2022</sup>
There is a new Material Icons collection called "Material Symbols"

https://fonts.google.com/icons?icon.set=Material+Symbols

As far as I understand it, there are a few more icons, some look different and also they are much more customizable than the old icons.

Also it seems to finally fix all those weird inconsistencies like having a filled "help" icon inside the "outlined" font :D
</div>

### PR DESCRIPTION
**Pull request by hanneskuettner**
## Description

Migrate over to Material Symbols, the never version of Material Icons that has ~1k new icons.  

Please see my [discussion comment](https://github.com/directus/directus/discussions/12958#discussioncomment-5478815) in #12958 for some background on the implications this change has.
TLDR: The font defines a lot of alternate codepoints that make it mostly backwards compatible. 

The `material-symbols.woff2` is a instanced version of the official Material Symbols Outline font that I created by running

`fonttools varLib.instancer material-symbols-outlined.woff2 wght=400 GRAD=0 opsz=24 FILL=0:0:1`

essentially removing the variability from the font by pinning the `wght`, `GRAD` and `opsz` axes to fixed values that correspond to the previously used fonts. This reduced the file size from 3MB to a much more manageable ~300kb. 

A total of 991 icons have been added with a lot of fallbacks for deprecated icon names. 
A total of 21 icons have been removed without any replacement.

<details><summary>List of removals</summary>

```
catching_pokemon
delivery_dining
discount
emoji_events
fitbit
highlight_alt
miscellaneous_services
moped
motion_photos_on
no_cell
panorama_horizontal_select
panorama_photosphere_select
panorama_vertical_select
panorama_wide_angle_select
person_add_alt_1
person_remove_alt_1
pix
play_circle_filled
play_circle_outline
signal_wifi_statusbar_connected_no_internet_4
whatsapp
```

</details>

The removals impacted the app only slightly. While crawling the used icons in the app I discovered 3 icons that were broken after the switch: 
* `bookmark_outline` (was already a undocumented alternate glyph and was officially called `bookmark_border` already in Material Icons and is now `bookmark`)
* `copy` (another undocumented glyph, was officially called `content_copy` already in Material Icons and is now removed and should be replaced)
* `highlight_alt` (has been removed, closest replacement I could find was [`select`](https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Aselect%3AFILL%400%3Bwght%40400%3BGRAD%400%3Bopsz%4048)

Another step should IMHO be replacing all the deprecated icons that are only there because of fallbacks with their canonical names. I will open another PR for that.

image

Fixes #18037

## Type of Change

- [ ] Bugfix
- [x] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [ ] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

**Comment by hanneskuettner** [comment]
I've got a question about migrations. Right now the soon to be removed icon `bookmark_outline` is used in a migration as a default for an icon field. I guess this would warrant another migration that alters the default?

https://github.com/directus/directus/blob/953c2f95d57a944ccb4771e41505b8d5caa80189/api/src/database/migrations/20220308A-add-bookmark-icon-and-color.ts#L4-L7

Any hint how to approach this? My educated guess is 

```ts
// 2023XXXXA-update-default-value-bookmark-icon.ts
export async function up(knex: Knex): Promise<void> {
	const helper = getHelpers(knex).schema;

	await helper.changeToType('directus_presets', 'icon', 'string', {
		nullable: true,
		default: 'bookmark',
		length: 30,
	});
}

export async function down(knex: Knex): Promise<void> {
	const helper = getHelpers(knex).schema;

	await helper.changeToType('directus_presets', 'icon', 'string', {
		nullable: true,
		default: 'bookmark_outline',
		length: 30,
	});
}
```

**Comment by rijkvanzanten** [comment]
@hanneskuettner Your educated guess is exactly right!

File: api/src/database/migrations/20230401B-rename-bookmark-icon-in-presets.ts
Comment (hanneskuettner) [inline api/src/database/migrations/20230401B-rename-bookmark-icon-in-presets.ts:1]: I also had to add a migration for updating the `bookmark_outline` to the `bookmark_border` icon in the `directus_presets` table since there a probably quite a few `bookmark_outline` values out there since it was the default icon (even though it didn't officially exist in the Material Icons).

I changed it to `bookmark_border` which isn't officially in the icon set (it's been replaced by `bookmark`) to allow for a clean downgrade. Since the user might have some presets with the old `bookmark` icon set, which is filled, and we probably don't want migrate that away.

File: app/src/interfaces/file-image/file-image.vue [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 		<div v-else-if="image" class="image-preview" :class="{ 'is-svg': image.type && image.type.includes('svg') }">
 			<div v-if="imageError || !src" class="image-error">
-				<v-icon large :name="imageError === 'UNKNOWN' ? 'error_outline' : 'info_outline'" />
+				<v-icon large :name="imageError === 'UNKNOWN' ? 'error_' : 'info'" />
```
Comment (rijkvanzanten) [inline app/src/interfaces/file-image/file-image.vue:11]:
```suggestion
				<v-icon large :name="imageError === 'UNKNOWN' ? 'error' : 'info'" />
```

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]
Out of 2837 changes in 54 files, I only found one typo! Good job @hanneskuettner 🎉 

I think we can merge the two migrations together, but other than that we should be in a good spot (i'll handle that tweak 👍🏻 )

File: app/src/interfaces/select-icon/select-icon.vue [hunk 2]
```diff
@@ -28,7 +28,10 @@
 		</template>
 
 		<div class="content" :class="width">
-			<template v-for="(group, index) in filteredIcons" :key="group.name">
+			<template v-for="group in filteredIcons" :key="group.name">
+				<v-divider v-if="group.icons.length > 0">
+					{{ group.name }}
+				</v-divider>
```
Comment (rijkvanzanten) [inline app/src/interfaces/select-icon/select-icon.vue:34]: Lets omit the name here to keep things the same as before. If we wanna show the group names, they have to be localized

**Review by rijkvanzanten — APPROVED** [review]
